File: CodeforcesCodes/R871D4D.cpp
#include<bits/stdc++.h>
using namespace std;

bool fct(int n, int m){
    if(n<m)
        return false;
    if(n==m)
        return true;
    if(n%3!=0)
        return false;
    if(fct(n/3, m))
        return true;
    if(fct((n/3)*2, m))
        return true;
    return false;
}
void solve(){
    int n;
    int a,b;
    cin>>a>>b;
    if(fct(a, b)){
        cout<<"YES"<<endl;
    }
    else{
        cout<<"NO"<<endl;
    }
}

int main(){
    int t;
    cin>>t;
    while(t--){
        solve();
    }
    return 0;
}

File: CodeforcesCodes/R871D4D.py
t = int(input())

for _ in range(t):
    def fct(n:int, m:int)->bool:
        if n<m:
            return False
        if n==m:
            return True
        if n%3!=0:
            return False
        if fct(n//3, m):
            return True
        if fct((n//3)*2, m):
            return True
        return False
    n, m = [int(i) for i in input().split()]
    if fct(n, m):
        print("YES")
    else:
        print("NO")